apps/src/applab/dataBrowser/DataOverview.jsx

```diff
@@ -125,9 +81,28 @@ const DataOverview = React.createClass({
 });
 
 export default connect(state => ({
-  view: state.data.view
+  view: state.data.view,
+  tableListMap: state.data.tableListMap || {}
 }), dispatch => ({
   onViewChange(view, tableName) {
     dispatch(changeView(view, tableName));
+  },
+  onTableAdd(tableName) {
+    // Add the table name to the model even though it doesn't exist in the database yet,
+    // so that it appears in the overview even if we don't create a record in it.
+    // addTableName() is idempotent, so we won't double-count the table name later
+    // when the the first record is added to it.
+    dispatch(addTableName(tableName));
+
+    dispatch(changeView(DataView.TABLE, tableName));
```

Comment:
Not too sure what I think about having our `mapDispatchToProps` method here creating a prop that dispatches two actions. Seems like should either have a single `tableAdd` action that takes care of everything, or we should have two props, and callers in our component call both `this.addTableName` and `this.changeView`.

My understanding of this method `mapDispatchToProps` (i.e. the second param passed to connect) is that it's mostly just to make it so that our component doesnt need to be calling dispatch directly. Additional complexity should be avoided (though you could make the case that this is still a small enough amount of complexity).